x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -0,0 +1,288 @@
+  ) {
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'Synthetics Sync Global Params Task',
+        description:
+          'This task is executed so that we can sync private location monitors for example when global params are updated',
+        timeout: '3m',
+        maxAttempts: 3,
+        createTaskRunner: (taskInstance) => {
+          return {
+            run: async () => {
+              return this.runTask(taskInstance);
+            },
+          };
+        },
+      },
+    });
+  }
+
+  public async runTask(context: RunContext) {
+    const {
+      coreStart: { savedObjects },
+      encryptedSavedObjects,
+      logger,
+    } = this.serverSetup;
+    const lastStartedAt =
+      context.taskInstance.state?.lastStartedAt || moment().subtract(10, 'minute');
+    const startedAt = context.taskInstance.startedAt || moment();
+    let lastTotalParams = context.taskInstance.state?.lastTotalParams || 0;
+    try {
+      logger.debug(
+        `Syncing private location monitors, last total params ${lastTotalParams}, last run ${lastStartedAt}`
+      );
+      const soClient = savedObjects.createInternalRepository();
+      const allPrivateLocations = await getPrivateLocations(soClient);
+      const { updatedParams, totalParams } = await this.hasAnyParamChanged(soClient, lastStartedAt);
+      if (updatedParams > 0 || totalParams !== lastTotalParams) {
+        lastTotalParams = totalParams;
+        logger.debug(
+          `Syncing private location monitors because params changed, updated params ${updatedParams}, total params ${totalParams}`
+        );
+
+        if (allPrivateLocations.length > 0) {
+          await this.syncGlobalParams({
+            allPrivateLocations,
+            soClient,
+            encryptedSavedObjects,
+          });
+        }
+        logger.debug(`Sync of private location monitors succeeded`);
+      } else {
+        lastTotalParams = totalParams;
+        logger.debug(
+          `No params changed since last run ${lastStartedAt}, skipping sync of private location monitors`
+        );
+      }
+    } catch (error) {
+      logger.error(`Sync of private location monitors failed: ${error.message}`);
+      return {
+        error,
+        state: {
+          lastStartedAt: startedAt.toISOString(),
+          lastTotalParams,
+        },
+      };
+    }
+    return {
+      state: {
+        lastStartedAt: startedAt.toISOString(),
+        lastTotalParams,
+      },
+    };
+  }
+
+  start = async () => {
+    const {
+      logger,
+      pluginsStart: { taskManager },
+    } = this.serverSetup;
+    logger.debug(`Scheduling private location task`);
+    await taskManager.ensureScheduled({
+      id: TASK_ID,
+      state: {},
+      schedule: {
+        interval: '10m',
+      },
+      taskType: TASK_TYPE,
+      params: {},
+    });
+    await taskManager.runSoon(TASK_ID);
+    logger.debug(`Sync private location monitors task scheduled successfully`);
+  };
```

Comment:
Do we want to run the task every 10m? Or is it supposed to be a ad-hod task scheduled manually when managing params?